include/sddf/i2c/queue.h

```diff
@@ -85,148 +141,185 @@ typedef struct i2c_queue_handle {
  * @param request pointer to request queue in shared memory.
  * @param response pointer to response queue in shared memory.
  */
-static inline i2c_queue_handle_t i2c_queue_init(i2c_queue_t *request, i2c_queue_t *response)
+static inline i2c_queue_handle_t i2c_queue_init(i2c_request_queue_t *request, i2c_response_queue_t *response)
 {
     i2c_queue_handle_t handle;
     handle.request = request;
     handle.response = response;
+    handle.request->staged_active_tail = 0;
 
     return handle;
 }
 
 /**
- * Check if the queue is empty.
+ * Check if the queue is empty by looking at all committed
+ * entries.
  *
  * @param queue queue to check.
  *
  * @return true indicates the queue is empty, false otherwise.
  */
-static inline int i2c_queue_empty(i2c_queue_t *queue)
+static inline int i2c_queue_empty(i2c_queue_ctrl_t q_ctrl)
 {
-    return queue->tail - queue->head == 0;
+    return q_ctrl.tail - q_ctrl.head == 0;
 }
 
 /**
- * Check if the queue is full
+ * Check if the queue is full by looking at all commited entries.
+ * Use i2c_request_queue_full() to consider uncommitted entries.
  *
  * @param queue queue to check.
  *
  * @return true indicates the buffer is full, false otherwise.
  */
-static inline int i2c_queue_full(i2c_queue_t *queue)
+static inline int i2c_queue_full(i2c_queue_ctrl_t q_ctrl)
+{
+    return q_ctrl.tail - q_ctrl.head + 1 == I2C_QUEUE_SZ;
+}
```

Comment:
Shouldn't this be `queue->capacity`? Like our other queues.

Reply:
I2C queues do not include a capacity like this since they are all equally sized

Reply:
If you look at the other queues you’ll see we use capacity instead of size for describing the maximum amount of elements in the queue, we should be consistent with those